hathor/transaction/transaction_metadata.py

```diff
@@ -109,6 +109,10 @@ class TransactionMetadata:
     first_block: Optional[bytes]
     height: int
     validation: ValidationState
+    # XXX: this is only used to defer the reward-lock verification from the transaction spending a reward to the first
+    # block that confirming this transaction, it is important to always have this set to be able to distinguish an old
+    # metadata (that does not have this calculated, from a tx with a new format that does have this calculated)
+    min_height: int
```

Comment:
Shouldn't we use `None` to distinguish from an old metadata? How do we know whether it is really zero or non-calculated?

Reply:
It should never be zero on this implementation.